### ISSUE TITLE
Uniform external electric fields

### ISSUE THREAD

**Issue by awvwgk** [grimme-lab/xtb#446]
**Is your feature request related to a problem? Please describe.**
Allow applying uniform external electrical fields to tight binding calculations.

**Describe the solution you'd like**
- allow adding an external uniform field as vector.
- apply potential of the field vector to atomic partial charges
- apply potential of the field vector to the atomic multipole moments

**Additional context**
Add any other context about the feature request here.

**Comment by gabegomes** [comment]
Not OP, but I must say that allowing to add an external uniform field as vector would be fantastic for my research.

**Comment by cmueli** [comment]
I strongly support this request, would be happy to have the option to include a homogeneuous electric field to the xTB calculations.

**Comment by awvwgk** [comment]
Since DFTB+ now supports the xTB Hamiltonian and already has support for electric fields I'll check whether we can already use this for calculations.

**Comment by vladislavivanistsev** [comment]
Will the external field be implemented in xTB package?

**Comment by awvwgk** [comment]
In the future, yes. We still have to decide on how those will be specified in `xtb`, since the input in DFTB+ is quite sophisticated (*e.g.* supporting time varying electric fields for dynamics).

A static, homogeneous electric field should be straight-forward to add in `xtb` now, if this is sufficient for most applications.

**Comment by vladislavivanistsev** [comment]
A homogeneous electric field is what needed to mimic electrified interface. I am testing DFTB+, which indeed has a sophisticated as well as complicated way of writing inputs. That is why I am more interested in using XTB or TBLITE directly or via ASE.

**Comment by awvwgk** [comment]
We are trying to improve the userfriendlyness of DFTB+, especially the input, but we probably won't get it on par with the command-line experience in `xtb`, it is kind of a different philosophy.

I'll look into porting the uniform fields to `xtb`.

**Comment by longkunxuluke** [comment]
Besides uniform electric field, implementing more advanced approaches (e.g., constant potential method) to model electrochemical interface would be highly desired as well. An example in LAMMPS: https://github.com/zhenxingwang/lammps-conp

**Comment by JOEYIT** [comment]
I have noticed that when i search the external potential in the xtb documentation. There is an example about the externel potential for the input script of water_4.pc. May I ask if it is the right way to set the externel electric field in xtb. As shown in the water_4.pc,    
 6
  -0.69645733    2.75237178376284    -2.43247309226225    -0.01392519847964  O
   0.36031084    0.93157260886974    -2.79621404458590    -0.01863384029005  H
   0.33614649    3.43820531288547    -3.30583608421060     1.42134539425148  H
  -0.69645733    2.43247309226225     2.75237178376284     0.01392519847964  O
   0.36031084    2.79621404458590     0.93157260886974     0.01863384029005  H
   0.33614649    3.30583608421060     3.43820531288547    -1.42134539425148  H
and 
 PC potential                     true          :
          :  -> # point charges                  6          :
          :  -> sum of PC                0.0000000     e   
if this 6 indicates the ev/Angstrom uniform externel electric field in the system?

**Comment by marcelmbn** [comment]
Quick update on this topic: The incorporation of static electric fields is possible within `xtb` with `--efield "x y z"` but only for GFN-FF and PTB, and not for the TB methods (as originally indicated in this thread).

**Comment by awvwgk** [comment]
This is currently supported with `tblite` and available in `xtb` (via `--tblite` option). Might need to check whether `xtb` correctly hands over the electric field to `tblite`.

**Comment by icamps** [comment]
It will be possible to use all the other modules like docking (aISS) and molecular dynamics?

**Comment by awvwgk** [comment]
Molecular dynamics should work out of the box. Not sure about aiss to be honest.

**Comment by icamps** [comment]
> Quick update on this topic: The incorporation of static electric fields is possible within `xtb` with `--efield "x y z"` but only for GFN-FF and PTB, and not for the TB methods (as originally indicated in this thread).

About PTB: I searched the xTB doc site, and didn't find anything related. The only mention to "Density Matrix Tight-Binding Potential" is at the end of the DIPRO section.

**Comment by marcelmbn** [comment]
> > Quick update on this topic: The incorporation of static electric fields is possible within `xtb` with `--efield "x y z"` but only for GFN-FF and PTB, and not for the TB methods (as originally indicated in this thread).
> 
> About PTB: I searched the xTB doc site, and didn't find anything related. The only mention to "Density Matrix Tight-Binding Potential" is at the end of the DIPRO section.

`xtb <structure_file> --ptb --efield "0.0 0.0 0.1"`
Documentation under https://xtb-docs.readthedocs.io/en/latest/index.html is on the way.

**Comment by TeddyHerriman** [comment]
Can I ask about the status of the documentation and implementation of external electric fields? I'm hoping to apply it within CREST.

**Comment by chselz** [comment]
With #1397 i think this is now done right? @thfroitzheim

**Comment by thfroitzheim** [comment]
Yes, the xtb Hamiltonians can now also include external electric fields if one uses the tblite implementation.
